custom/icds_reports/tasks.py

```diff
@@ -1209,6 +1249,114 @@ def build_incentive_report(agg_date=None):
                 build_incentive_files.delay(location, agg_date, file_format, 3, location.parent.parent, location.parent)
 
 
+# performance row is AWCIncentiveReport object with 'awc_id', 'is_launched', 'valid_visits',
+# 'visit_denominator', 'wer_weighed', 'wer_eligible', 'incentive_eligible'
+# awcagg_row is AggAwc object with 'awc_id', 'is_launched', 'valid_visits', 'expected_visits'
+def get_awc_is_launched_mismatch_row(performance_row, awcagg_row):
+    is_launched_from_awc = False
+    if awcagg_row[1].lower() == 'yes':
+        is_launched_from_awc = True
+    # checking if is_launched from incentive report is same as that of aggregate
+    if performance_row[1] != is_launched_from_awc:
+        row_data = [performance_row[0], awcagg_row[0], performance_row[1], is_launched_from_awc]
+        return row_data
+    return None
+
+
+# performance row is AWCIncentiveReport object with 'awc_id', 'is_launched', 'valid_visits',
+# 'visit_denominator', 'wer_weighed', 'wer_eligible', 'incentive_eligible'
+# awcagg_row is AggAwc object with 'awc_id', 'is_launched', 'valid_visits', 'expected_visits'
+def get_awc_home_conduct_mismatch_row(performance_row, awcagg_row):
+    # checking if valid_visits or valid_denominator is zero or None as they are treated as valid cases
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
+        home_conduct_from_report = 'None'
+    else:
+        home_conduct_from_report = performance_row[2] / performance_row[3]
+    if awcagg_row[2] in [0, None] or awcagg_row[3] in [0, None]:
+        home_conduct_from_awc = 'None'
+    else:
+        home_conduct_from_awc = awcagg_row[2] / awcagg_row[3]
+    # checking if home conduct (valid_visits/valid_denominator) from incentive report is same as that of aggregate
+    if home_conduct_from_report != home_conduct_from_awc:
+        row_data = [performance_row[0], awcagg_row[0], home_conduct_from_report, home_conduct_from_awc]
+        return row_data
+    return None
+
+
+# performance row is AWCIncentiveReport object with 'awc_id', 'is_launched', 'valid_visits',
+# 'visit_denominator', 'wer_weighed', 'wer_eligible', 'incentive_eligible'
+def get_awc_eligibility_mismatch_row(performance_row):
+    # checking if valid_visits or valid_denominator is zero or None as they are treated as valid cases
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
+        is_eligible = True
+    else:
+        # checking if valid_visits/valid_denominator > 0.6(60%)
+        home_conduct_from_report = performance_row[2] / performance_row[3]
+        if home_conduct_from_report > 0.6:
+            is_eligible = True
+        else:
+            is_eligible = False
+    if is_eligible:
+        # checking if wer_weighed or wer_eligible is zero or None as they are treated as valid cases
+        if performance_row[4] in [0, None] or performance_row[5] in [0, None]:
```

Comment:
same comment here as above 0/0 counts. 0/5 does not

Reply:
got it